cmd/nerdctl/images.go

```diff
@@ -143,19 +146,40 @@ func imagesBashComplete(clicontext *cli.Context) {
 	bashCompleteImageNames(clicontext)
 }
 
-func unpackedImageSize(ctx context.Context, clicontext *cli.Context, client *containerd.Client, i images.Image) (int64, error) {
+type snapshotKey map[string]string
+
+func unpackedImageSize(ctx context.Context, clicontext *cli.Context, client *containerd.Client, s snapshots.Snapshotter, i images.Image) (int64, error) {
 	img := containerd.NewImage(client, i)
 
 	diffIDs, err := img.RootFS(ctx)
 	if err != nil {
 		return 0, err
 	}
+
 	chainID := identity.ChainID(diffIDs).String()
-	s := client.SnapshotService(clicontext.String("snapshotter"))
 	usage, err := s.Usage(ctx, chainID)
 	if err != nil {
 		return 0, err
 	}
 
+	keys := make(snapshotKey, 0)
+	walkFn := func(ctx context.Context, info snapshots.Info) error {
+		keys[info.Name] = info.Parent
+		return nil
+	}
+	s.Walk(ctx, walkFn)
+
+	//add ChainID's parent usage to the total usage
+	keys.add(ctx, s, &usage, keys[chainID])
 	return usage.Size, nil
 }
+
+// recursive function to calculate total usage of ChainID's parent
+func (keys snapshotKey) add(ctx context.Context, s snapshots.Snapshotter, usage *snapshots.Usage, key string) {
+	if key == "" {
+		return
+	}
+	u, _ := s.Usage(ctx, key)
+	usage.Add(u)
+	keys.add(ctx, s, usage, keys[key])
+}
```

Comment:
This (`snapshotKey.add`) should be put immediately following the type definition of `snapshotKey` (`type snapshotKey map[string]string`) .